geometry/polygon.hpp

```diff
@@ -99,6 +113,29 @@ double area( const std::vector<math::Point2_<T> > & polygon  ) {
     return 0.5 * retval;
 }
 
+template <typename T>
+double areaAndCentroid(math::Point2_<T>& centroid,
+                       const std::vector<math::Point2_<T>>& polygon)
+{
+    auto n = polygon.size();
+    centroid = math::Point2_<T>(0, 0);
+    double a = 0.0;
+
+    for ( decltype(n) i = 0; i < n; i++ ) {
+        decltype(n) iN = (i + 1) % n;
```

Comment:
Why did we choose to use `decltype` here while in the other functions we are using `std::size_t`? Even if this was needed a better approach is to get the size_type at the beginning of the function; e.g.
```
using size_type = std::vector<math::Point2_<T>>::size_type;
```
And then use it in the rest of the function:
```
const size_type n { polygon.size() };
centroid = math::Point2_<T>(0, 0);
double a = 0.0;

for (size_type i { 0 }; i < n; ++i)
{
    const size_type iN { (i + 1) % n };
```

Reply:
I followed the convention in 
https://github.com/melowntech/libgeometry/blob/9d804baa191213e2bd9e99c5d27ebd5b203b4571/geometry/polygon.hpp#L106

Introduced in https://github.com/melowntech/libgeometry/commit/6e71f6e091bf2d7789245f74ce1ae47c1accb800 

But it seems that there is no reason not to use `std::size_t` and I like it better so I changed it...